src/dotenv/main.py

```diff
@@ -90,24 +115,33 @@ def parse_stream(stream):
 
 class DotEnv():
 
+    if TYPE_CHECKING:
+        dotenv_path = None  # type: Union[Text,_PathLike, _StringIO]
+        _dict = None  # type: Optional[Dict[Text, Text]]
+        verbose = None  # type: bool
```

Comment:
I think you can move these type hints to the assignments below in `__init__`.  This removes the need for `if TYPE_CHECKING` and is probably more accurate since those aren't class variables.